tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go

```diff
@@ -0,0 +1,131 @@
+			})
+
+			By("Deleting rollback cgu created on target hub cluster", func() {
+				err := cnfhelper.DeleteIbuTestCguOnTargetHub(cnfinittools.TargetHubAPIClient, tsparams.RollbackCguName,
+					tsparams.IbuCguNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to delete rollback cgu on target hub cluster")
+			})
+		})
+
+		It("Rollback after failed upgrade", reportxml.ID("69054"), func() {
+			By("Creating, enabling ibu pre-prep CGU and waiting for CGU status to report completed", func() {
+				prePrepCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
+					tsparams.PrePrepCguName, tsparams.IbuCguNamespace, 1).
+					WithCluster(tsparams.TargetSnoClusterName).
+					WithManagedPolicy(tsparams.PrePrepPolicyName).
+					WithCanary(tsparams.TargetSnoClusterName)
+				prePrepCguBuilder.Definition.Spec.Enable = ptr.To(true)
+
+				prePrepCguBuilder, err := prePrepCguBuilder.Create()
+				Expect(err).ToNot(HaveOccurred(), "Failed to create pre-prep CGU.")
+
+				_, err = prePrepCguBuilder.WaitUntilComplete(10 * time.Minute)
+				Expect(err).ToNot(HaveOccurred(), "Pre-prep CGU did not complete in time.")
+			})
+
+			By("Creating, enabling ibu prep CGU and waiting for CGU status to report completed", func() {
+				prepCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
+					tsparams.PrepCguName, tsparams.IbuCguNamespace, 1).
+					WithCluster(tsparams.TargetSnoClusterName).
+					WithManagedPolicy(tsparams.PrepPolicyName).
+					WithCanary(tsparams.TargetSnoClusterName)
+				prepCguBuilder.Definition.Spec.Enable = ptr.To(true)
+
+				prepCguBuilder, err := prepCguBuilder.Create()
+				Expect(err).ToNot(HaveOccurred(), "Failed to create prep CGU.")
+
+				_, err = prepCguBuilder.WaitUntilComplete(25 * time.Minute)
+				Expect(err).ToNot(HaveOccurred(), "Prep CGU did not complete in time.")
+			})
+
+			By("Creating, and enabling ibu upgrade CGU", func() {
+				upgradeCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
+					tsparams.UpgradeCguName, tsparams.IbuCguNamespace, 1).
+					WithCluster(tsparams.TargetSnoClusterName).
+					WithManagedPolicy(tsparams.UpgradePolicyName).
+					WithCanary(tsparams.TargetSnoClusterName)
+				upgradeCguBuilder.Definition.Spec.Enable = ptr.To(true)
+
+				_, err := upgradeCguBuilder.Create()
+				Expect(err).ToNot(HaveOccurred(), "Failed to create upgrade CGU.")
+			})
+
+			By("Performing rollback after a failed upgrade", func() {
+				upgradeStagePolicyStatusCheck, err := ocm.PullPolicy(cnfinittools.TargetHubAPIClient,
+					tsparams.UpgradePolicyName,
+					tsparams.IbuPolicyNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to pull upgrade stage policy")
+
+				err = upgradeStagePolicyStatusCheck.WaitUntilComplianceState(policiesv1.Compliant, 40*time.Minute)
+				Expect(err).ToNot(HaveOccurred(), "Upgrade-stage-policy failed to report Compliant state")
+
+				if err != nil {
```

Comment:
`err` would always be `nil` here since you're using `Expect(err).ToNot(HaveOccurred())` above it, did you want to assert that the error did occur here?

Reply:
Thanks. I have addressed it and updated the changes.